error: too short salt during C compatibility test. now requires 8 chars

- Compatibility test in PR https://github.com/KEINOS/go-argonize/pull/57 is failing as below:
  - https://github.com/KEINOS/go-argonize/actions/runs/13249874224/job/36985047322?pr=57#step:13:9

- Reason:
  - Currently, `argon2` requires 8 char length of salt but the [test uses 6 chars](https://github.com/KEINOS/go-argonize/actions/runs/13249874224/job/36985047322#step:13:8).

  ```shellsession
  #  ./argon2 -h
  Usage:  ./argon2 [-h] salt [-i|-d|-id] [-t iterations] [-m log2(memory in KiB) | -k memory in KiB] [-p parallelism] [-l hash length] [-e|-r] [-v (10|13)]
  	Password is read from stdin
  Parameters:
  	salt		The salt to use, at least 8 characters
  	-i		Use Argon2i (this is the default)
  	-d		Use Argon2d instead of Argon2i
  	-id		Use Argon2id instead of Argon2i
  	-t N		Sets the number of iterations to N (default = 3)
  	-m N		Sets the memory usage of 2^N KiB (default 12)
  	-k N		Sets the memory usage of N KiB (default 4096)
  	-p N		Sets parallelism to N threads (default 1)
  	-l N		Sets hash output length to N bytes (default 32)
  	-e		Output only encoded hash
  	-r		Output only the raw bytes of the hash
  	-v (10|13)	Argon2 version (defaults to the most recent version, currently 13)
  	-h		Print ./argon2 usage
  ```

- Cause:

https://github.com/KEINOS/go-argonize/blob/6b0011a9742646d4ad09f655b25909785b871563/_tests/main.sh#L14-L21

- Fix:
  
  ```diff
   # Generate a random salt 
  -  salt="${RANDOM}${RANDOM}" 
  +  salt="${RANDOM}${RANDOM}${RANDOM}" # 8 char rand num
   echo "Salt: ${salt} (random)" 
    
   # Generate a hash using the C implementation 
   # shellcheck disable=SC3037 # POSIX sh, echo -n flag is not supported but ash in Alpine does 
   hashClang=$(echo -n "${myPassword}" | argon2 "${salt}" -t 1 -m 16 -p 2 -l 32 -id | grep Encoded | sed -r 's/Encoded:(.)/\2/') 
   echo "Hash (Clang ): ${hashClang}" 
  ```

--- a/.github/Dockerfile
+++ b/.github/Dockerfile
@@ -1,8 +1,8 @@
 # =============================================================================
 #  Test Container for Vaious Go Versions (Last Updated: 2022-08-05)
 # =============================================================================
-# Default version: golang:alpine
-ARG VARIANT="alpine"
+# Default version: minimum supported go version
+ARG VARIANT="1.23-alpine"
 
 # -----------------------------------------------------------------------------
 #  Main Stage

--- a/.github/docker-compose.yml
+++ b/.github/docker-compose.yml
@@ -62,7 +62,7 @@ services:
       dockerfile: ./_tests/Dockerfile_py
     volumes:
       - ../_tests/main.py:/workspaces/main.py
-  # Compatibility test with C
+  # Compatibility test with C (not the clang compiler)
   clang:
     build:
       context: ..

--- a/.github/docker-compose.yml
+++ b/.github/docker-compose.yml
@@ -18,13 +18,11 @@ volumes:
   GO_PKG_MOD:
 
 services:
-  # Service tidy updates the go.mod to the latest
+  # Service tidy updates the go.mod on minimum Go version
   tidy:
     build:
       context: ..
       dockerfile: ./.github/Dockerfile
-      args:
-        VARIANT: 1.22-alpine
     volumes:
       - ..:/workspaces
     entrypoint: [ "./.github/run_go_mod_tidy.sh" ]
@@ -33,8 +31,6 @@ services:
     build:
       context: ..
       dockerfile: ./.github/Dockerfile
-      args:
-        VARIANT: 1.22-alpine
     volumes:
       - ..:/workspaces
       - GO_PKG_MOD:/go/pkg/mod

--- a/.github/run_go_mod_tidy.sh
+++ b/.github/run_go_mod_tidy.sh
@@ -13,21 +13,14 @@
 
 set -eu
 
+min_go_version="1.23.0"
+
 echo '* Backup modules ...'
 cp -f go.mod go.mod.bak
 cp -f go.sum go.sum.bak
 
-# echo '* Create new blank go.mod ...'
-# # Copy the first 4 lines of the go.mod.bak file to the new go.mod file.
-# < go.mod.bak head -n 4 > go.mod
-
-# echo '* Get latest modules ...'
-# go get "github.com/pkg/errors"
-# go get "github.com/stretchr/testify"
-# go get "golang.org/x/crypto"
-
 echo '* Run go tidy ...'
-go mod tidy -go=1.22
+go mod tidy -go=$min_go_version
 
 echo '* Run tests ...'
 go test ./... && {

--- a/.github/workflows/code-coverage.yml
+++ b/.github/workflows/code-coverage.yml
@@ -17,7 +17,7 @@ jobs:
 
       - uses: actions/setup-go@v5
         with:
-          go-version: '1.22'
+          go-version: '1'
           check-latest: true
 
       - name: Run coverage

--- a/.github/workflows/unit-tests.yml
+++ b/.github/workflows/unit-tests.yml
@@ -53,13 +53,8 @@ jobs:
       - name: Pull base images if no-cache
         if: steps.cache.outputs.cache-hit != 'true'
         run: |
-          : # Pull minimum supported Go version
-          docker pull golang:1.22-alpine
-          : # Pull latest Go version
-          docker pull golang:alpine
-          : # Pull latest PHP and Python images
-          docker pull php:alpine
-          docker pull python:alpine
+          # Pull the base images for stability
+          docker compose --file ./.github/docker-compose.yml pull
 
       - name: Build Docker images if no-cache
         if: steps.cache.outputs.cache-hit != 'true'

--- a/README.md
+++ b/README.md
@@ -11,7 +11,7 @@ go get "github.com/KEINOS/go-argonize"
 ```
 
 ```go
-func Example() {
+func Example_basic() {
     // Your strong and unpredictable password
     password := []byte("my password")
 
@@ -45,11 +45,42 @@ func Example() {
 }
 ```
 
+```go
+func Example_from_saved_password() {
+  // Load the hashed password from a file, DB or etc.
+  savedPasswd := "$argon2id$v=19$m=65536,t=1,p=2$iuIIXq4foOhcGUH1BjE08w$kA+XOAMls8hzWg3J1sYxkeuK/lkU4HDRBf0zchdyllY"
+
+  // Decode the saved password to a Hashed object.
+  // Note that once hashed, passwords cannot be recovered and can only be
+  // verified.
+  hashObj, err := argonize.DecodeHashStr(savedPasswd)
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  // Validate the password against the hashed password.
+  if hashObj.IsValidPassword([]byte("my password")) {
+    fmt.Println("the password is valid")
+  } else {
+    fmt.Println("the password is invalid")
+  }
+
+  if hashObj.IsValidPassword([]byte("wrong password")) {
+    fmt.Println("the password is valid")
+  } else {
+    fmt.Println("the password is invalid")
+  }
+  // Output:
+  // the password is valid
+  // the password is invalid
+}
+```
+
 - [View more examples and advanced usages](https://pkg.go.dev/github.com/KEINOS/go-argonize#pkg-examples) @ pkg.go.dev
 
 ## Contributing
 
-[![go1.22+](https://img.shields.io/badge/Go-1.22+-blue?logo=go)](https://github.com/KEINOS/go-argonize/blob/main/.github/workflows/unit-tests.yml#L81 "Supported versions")
+[![GitHub go.mod Go version](https://img.shields.io/github/go-mod/go-version/KEINOS/go-argonize)](https://github.com/KEINOS/go-argonize/blob/main/.github/workflows/unit-tests.yml#L81 "Supported versions")
 [![Go Reference](https://pkg.go.dev/badge/github.com/KEINOS/go-argonize.svg)](https://pkg.go.dev/github.com/KEINOS/go-argonize/ "View document")
 [![Opened Issues](https://img.shields.io/github/issues/KEINOS/go-argonize?color=lightblue&logo=github)](https://github.com/KEINOS/go-argonize/issues "opened issues")
 [![PR](https://img.shields.io/github/issues-pr/KEINOS/go-argonize?color=lightblue&logo=github)](https://github.com/KEINOS/go-argonize/pulls "Pull Requests")

--- a/_tests/main.sh
+++ b/_tests/main.sh
@@ -12,7 +12,7 @@ myPassword="$(mkpasswd -m sha512crypt "${RANDOM}" | tr -d "$/.")"
 echo "Password: ${myPassword} (random)"
 
 # Generate a random salt
-salt="${RANDOM}${RANDOM}"
+salt="${RANDOM}${RANDOM}${RANDOM}" # 8 char rand num (fix #58)
 echo "Salt: ${salt} (random)"
 
 # Generate a hash using the C implementation

--- a/go.mod
+++ b/go.mod
@@ -1,7 +1,8 @@
 module github.com/KEINOS/go-argonize
 
-go 1.22
-toolchain go1.23.1
+go 1.23.0
+
+toolchain go1.24.0
 
 require (
 	github.com/pkg/errors v0.9.1

